### PR TITLE
Integrate researcher chat with API

### LINKED ISSUE #721: Integrar o chat com o frontend
## Feature Description

Com a finalização do backend (https://github.com/ufersa/plataforma-sabia/pull/676) do chat, precisamos integrar essa feature com o frontend, existe uma documentação na API também. 

---------------

## Implementation Brief

* Enviar o request para o endpoint a seguir para criar (caso seja a primeira iteração entre os users) ou apenas acessar (acesso seguintes)

target_user -> "dono" da tecnology-order que eu quero conversar

```
GET http://api-url/chat?object_type=technology-order&object_id=1&target_user=2
Authorization: bearer xxxxx
```

* Para fazer o scroll ou pegar as ultimas messages

```
GET http://api-url/chat/cc692a91-b8bd-4879-b7ae-57497a6db512/messages?offset=10
Authorization: bearer xxxxx
```

* Enviar uma mensagem no chat

```
POST http://api-url/chat/cc692a91-b8bd-4879-b7ae-57497a6db512/messages
Authorization: bearer xxxxxxx
Content-Type: application/json

{
    "content": {
        "text": "Bom dia, está disponível?"
    }
}
```

### PR DESCRIPTION
**Pull request by luizeboli**
## Summary

Resolves #721 

https://www.loom.com/share/2504c901f4e240898327ef094f6c66c2

## Relevant technical choices

* This PR only integrates researcher area because we didn't have a way to get current user orders. I'll open a new PR to solve it.
* I'll write cypress coverage when integrating user area chat
* I did some small refactoring to improve code reuse

## QA Steps

1. Watch above video 😄 
2. Login with admin user `sabiatestingadmin@gmail.com` (this is the one with all register fields filled, required to buy a technology)
3. Access `/t/5` page (or any technology that is open to buy) 
4. Click `Adquirir essa tecnologia` button
5. Fill all modal fields and click `Adquirir tecnologia` button
6. Login with `sabiatestinge2e@gmail.com` (this is the technology owner)
7. Access `Pedidos` menu in `Área do pesquisador`
8. Search for the order that you just made
9. Click the chat icon to open messages view
10. Send a message

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/web/components/OrderMessages/index.js [hunk 1]
```diff
@@ -130,14 +192,18 @@ const OrderMessages = ({ isBuyer, currentOrder, backToList }) => {
 
 					<div>
 						<img
-							src="https://rocketfinalchallenge.s3.amazonaws.com/card-image.jpg"
+							src={currentOrder.technology?.thumbnail?.url || '/card-image.jpg'}
 							alt="Technology thumbnail"
 						/>
```
Comment (mateus4k) [inline packages/web/components/OrderMessages/index.js:197]: Can we use Next's Image component here?

**Review by mateus4k — APPROVED** [review]
👍

**Review by calaca — APPROVED** [review]
@luizeboli awesome work!

Code looks great as always and the feature works just as expected.

I believe the workaround you have managed to create deals well with the case where the first message is loaded and the "load more" button is disabled. I do think it would be nice if we could add some pagination info to the API.

**Comment by luizeboli** [comment]
Totally agree @calaca, it would be really nice

**Pull request merged**